Tasks/AppCenterDistributeV2/appcenterdistribute.ts

```diff
@@ -88,69 +104,101 @@ function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: stri
     request.post({ url: beginUploadUrl, headers: headers }, (err, res, body) => {
         responseHandler(defer, err, res, body, () => {
             let response = JSON.parse(body);
-            let uploadInfo: UploadInfo = {
-                upload_id: response['upload_id'],
-                upload_url: response['upload_url']
+            if (!response.package_asset_id || (response.statusCode && (response.statusCode < 200 || response.statusCode >= 300))) {
+                defer.reject(`failed to create release upload. ${response.message}`)
             }
-
-            defer.resolve(uploadInfo);
+            defer.resolve(response);
         });
     });
-
     return defer.promise;
 }
 
-function uploadRelease(uploadUrl: string, file: string, userAgent: string): Q.Promise<void> {
-    tl.debug("-- Uploading release...");
+function abortReleaseUpload(apiServer: string, apiVersion: string, appSlug: string, upload_id: string, token: string, userAgent: string): Q.Promise<void> {
+    tl.debug("-- Aborting release...");
     let defer = Q.defer<void>();
-    tl.debug(`---- url: ${uploadUrl}`);
+    let patchReleaseUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/release_uploads/${upload_id}`;
+    tl.debug(`---- url: ${patchReleaseUrl}`);
     let headers = {
+        "X-API-Token": token,
         "User-Agent": userAgent,
         "internal-request-source": "VSTS"
     };
-    let req = request.post({ url: uploadUrl, headers: headers }, (err, res, body) => {
+    let abortedBody = { "status": "aborted" };
+    request.patch({ url: patchReleaseUrl, headers: headers, json: abortedBody }, (err, res, body) => {
         responseHandler(defer, err, res, body, () => {
-            tl.debug('-- File uploaded.');
+
+            const { _, message } = body;
```

Comment:
Do you need to extract '_' here?

Reply:
No, first parameter is upload_status, we don't need that for logging.

Reply:
So do we need to extract it here?